Package.swift

```diff
@@ -36,8 +36,10 @@ let package = Package(
         .library(name: "SmithyXML", targets: ["SmithyXML"]),
```

Comment:
Added 2 new modules: `SmithyIdentity` and `SmithyHTTPAuth`. 

`SmithyIdentity` has concrete implementation of the `Identity` protocol from `SmithyIdentityAPI` module, such as `AWSCredentialIdentity`. It also has non-AWS specific AWS credential identity resolver such as `StaticAWSCredentialIdentityResolver`.

`SmithyHTTPAuth` has SigV4 signing config things moved from aws-sdk-swift. Moving the actual `SigV4Signer` and the auth schemes (e.g., `SigV4AuthScheme`) is on-hold for after HTTP modularization.